### PR TITLE
Generate Namespace Registry files from CSV

### LINKED ISSUE #4: DOM Conversion
Problem: Input of two or more LDDs to LDDTool (an LDD stack) results in null pointer exception. 
Analysis: LDDTool tried to find LDD nameSpaceID for "Import" statements in configured PDS4 nameSpaceID array. 
Solution: Use masterAllSchemaFileSortMap instead of masterSchemaFileSortMap. The latter includes only those nameSpaces to be written to XML Schema file.

### LINKED ISSUE #1053: Automate update of namespace-registry PDF and XLSX when CSV is updated
### Are you sure this is not a new requirement or bug?

No

### Task Type

Sub-task

### 💡 Description

Recently, the update of docs/namespace-registry/pds-namespace-registry.csv has been automated, but the .pdf and .xlsx files remain untouched.

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary

On a push of `docs/namespace-registry/pds-namespace-registry.csv` to the `main` branch, a workflow is triggered that will generate (not update) corresponding PDF and XLSX files matching the style of the currently existing files.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 100%

## ♻️ Related Issues

Resolves https://github.com/NASA-PDS/pds4-information-model/issues/1053

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: .github/workflows/generate-namespace-registry-files.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # Fetch at least the immediate parents for PR context
+          fetch-depth: 2
+
+      - name: Install dependencies
+        run: |
+          pip install openpyxl reportlab
```
Comment (nutjob4life) [inline .github/workflows/generate-namespace-registry-files.yml]: No version numbers on `openpyxl` and `reportlab` make this vulnerable to a Supply Chain ([Dependency Confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)) Vulnerability. I strongly recommend adding pessimistic version constraints, like

    pip install --quiet 'openpyxl~=3.1.5' 'reportlab~=4.5.1'

File: .github/workflows/generate-namespace-registry-files.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if: steps.changes.outputs.has_changes == 'true'
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add docs/namespace-registry/pds-namespace-registry.xlsx docs/namespace-registry/pds-namespace-registry.pdf
+          git commit -m "Auto-generate namespace registry files from CSV [skip ci]
+
+          Generated by GitHub Actions
+
+          Co-Authored-By: Claude (claude-sonnet-4.5) <noreply@anthropic.com>"
```
Comment (nutjob4life) [inline .github/workflows/generate-namespace-registry-files.yml]: Wouldn't seeing this (possibly frequent) commit message get a little noisy? I'm not sure we need Claude's thumbprint on every one when "Auto-generate" tells all we really need to know, right?

**Review by nutjob4life — CHANGES_REQUESTED** [review]

**Comment by nutjob4life** [comment]
Good start! See above for just a few minor nits the pedantic side of me needs to pick 😁

**Comment by c-suh** [comment]
@nutjob4life I appreciate the picking! It refines the product and teaches me as well. I'm working on copilot's catches for the PY file, but I've addressed your 2 requests and marked those as resolved.

File: .github/workflows/generate-namespace-registry-files.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # Fetch at least the immediate parents for PR context
+          fetch-depth: 2
+
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml:26]: Ignoring to keep consistent with other workflows in this repo

File: .github/workflows/generate-namespace-registry-files.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Install dependencies
+        run: |
+          pip install openpyxl reportlab
+
+      - name: Run generation script
+        run: |
+          python scripts/generate-namespace-registry-files.py
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml:33]: Ignoring in favor of nutjob4life's comment and requested change.

**Comment by jordanpadams** [comment]
@c-suh can we update this PR title?

**Comment by c-suh** [comment]
@jordanpadams hah yes, done!

File: .github/workflows/generate-namespace-registry-files.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add docs/namespace-registry/pds-namespace-registry.xlsx docs/namespace-registry/pds-namespace-registry.pdf
+          git commit -m "Auto-generate namespace registry files from CSV [skip ci]
+
+          Generated by GitHub Actions
+
+          Co-Authored-By: Claude (claude-sonnet-4.5) <noreply@anthropic.com>"
+          git push
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml:51]: Ignoring as nutjob4life hasn't flagged this as an issue and because this run is contingent on the CSV file having been changed.

File: scripts/generate-namespace-registry-files.py [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+Generate PDS Namespace Registry files (PDF and XLSX) from CSV.
+
+This script reads pds-namespace-registry.csv and creates:
+- pds-namespace-registry.xlsx
+- pds-namespace-registry.pdf
+
+Requirements:
+    Python 3.13+
+    pip install openpyxl reportlab
+"""
```
[the remark this answers is not in the document]
Comment (c-suh) [reply scripts/generate-namespace-registry-files.py:12]: https://github.com/NASA-PDS/pds4-information-model/pull/1054/changes/b9b0b340d4eb112469117df2a5e970bac7e5c49f

File: scripts/generate-namespace-registry-files.py [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+        # Ensure we have exactly 7 columns
+        while len(csv_row) < 7:
+            csv_row.append('')
+
+        # Check if this row is a section header
+        is_section_header = False
+        if csv_row and csv_row[0].strip() in ['Common', 'International', 'Discipline', 'Mission', 'Held For Future Use']:
+            is_section_header = True
+            section_header_rows.append(row_idx)
+
```
[the remark this answers is not in the document]
Comment (c-suh) [reply scripts/generate-namespace-registry-files.py]: https://github.com/NASA-PDS/pds4-information-model/pull/1054/changes/b9b0b340d4eb112469117df2a5e970bac7e5c49f

File: scripts/generate-namespace-registry-files.py [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+def create_xlsx(csv_rows: list[list[str]], output_path: Path) -> None:
+    """Create an XLSX file with styling matching the actual file."""
+    wb = openpyxl.Workbook()
+    ws = wb.active
+    ws.title = "Sheet1"
+
+    # Define styles
+    # Header row style (rows 1 and 2) - dark gray background, bold, white text
+    header_font = Font(name='Calibri', size=12, bold=True, color="000000")
+    header_fill = PatternFill(start_color="D9D9D9", end_color="D9D9D9", fill_type="solid")
```
[the remark this answers is not in the document]
Comment (c-suh) [reply scripts/generate-namespace-registry-files.py:52]: https://github.com/NASA-PDS/pds4-information-model/pull/1054/changes/b9b0b340d4eb112469117df2a5e970bac7e5c49f

**Review by nutjob4life — APPROVED** [review]
Looking good! 

Delta: ✓
Subsequent deltas: ✓
Approval: ✅
